src/com/facebook/buck/android/DefaultAndroidLibraryCompilerFactory.java

```diff
@@ -16,23 +16,39 @@
 
 package com.facebook.buck.android;
 
+import com.facebook.buck.jvm.kotlin.KotlinBuckConfig;
 import com.facebook.buck.jvm.scala.ScalaBuckConfig;
-import com.facebook.buck.util.HumanReadableException;
+import com.facebook.buck.rules.PathSourcePath;
+import com.facebook.buck.rules.SourcePath;
+
+import com.google.common.io.Files;
 
 public class DefaultAndroidLibraryCompilerFactory implements AndroidLibraryCompilerFactory {
 
   private final ScalaBuckConfig scalaConfig;
+  private final KotlinBuckConfig kotlinBuckConfig;
 
-  public DefaultAndroidLibraryCompilerFactory(ScalaBuckConfig scalaConfig) {
+  public DefaultAndroidLibraryCompilerFactory(
+      ScalaBuckConfig scalaConfig, KotlinBuckConfig kotlinBuckConfig) {
     this.scalaConfig = scalaConfig;
+    this.kotlinBuckConfig = kotlinBuckConfig;
   }
 
   @Override
-  public AndroidLibraryCompiler getCompiler(AndroidLibraryDescription.JvmLanguage language) {
-    switch (language) {
-      case JAVA: return new JavaAndroidLibraryCompiler();
-      case SCALA: return new ScalaAndroidLibraryCompiler(scalaConfig);
+  public AndroidLibraryCompiler getCompiler(AndroidLibraryDescription.Arg args) {
+    // We currently expect all non-java languages to only contain files of that extension.
```

Comment:
Do we check this assumption anywhere? If not, we should, and throw a `HumanReadableException` that explains what is wrong.